client/rpc/src/eth.rs

```diff
@@ -821,17 +821,17 @@ impl<B, C, P, CT, BE, H: ExHashT> EthApiT for EthApi<B, C, P, CT, BE, H> where
 					}
 
 					Err(err) => {
-						// if Err == OutofGas or OutofFund, we need more gas
+						// if Err == OutofGas, we need more gas
 						if err.code == ErrorCode::ServerError(0) {
 							lower = mid;
 							mid = (lower + upper + 1) / 2;
 							if mid == lower {
 								break;
 							}
+						} else {
```

Comment:
Are we sure the logic here is correct? If error is not out-of-gas and `mid` not equal to `lower`, then it would skip returning `Err` and instead return `Ok`.

I'm not sure the intention here but this at least should be reverted or changed.

Reply:
Hmm sorry. I was confused. The previous logic seems wrong indeed.

Reply:
Yes, In previous logic, it will early return when `OutOfGas`.